Sources/Swift/Integrations/SessionReplay/SentryReplayOptions.swift

```diff
@@ -216,39 +278,108 @@ public class SentryReplayOptions: NSObject, SentryRedactOptions {
      *  - errorSampleRate Indicates the percentage in which a 30 seconds replay will be send with
      * error events.
      */
-    public init(sessionSampleRate: Float = 0, onErrorSampleRate: Float = 0, maskAllText: Bool = true, maskAllImages: Bool = true, enableViewRendererV2: Bool = false, enableFastViewRendering: Bool = false) {
+    public init(
+        sessionSampleRate: Float = 0,
+        onErrorSampleRate: Float = 0,
+        maskAllText: Bool = true,
+        maskAllImages: Bool = true,
+        enableViewRendererV2: Bool = true,
+        enableFastViewRendering: Bool = false
+    ) {
+        // Note for future maintainers:
+        // - This initializer is publicly available for Swift, but not for Objective-C.
+        // - Each parameter has a default value, so the parameter can be omitted.
+        // - The values are not optional, because SDK users should not be able to set them to nil.
+
+        // WHEN CHANGING ANY DEFAULT VALUE IN THE INITIALIZER, UPDATE THE PROPERTY DOCUMENTATION
+
         self.sessionSampleRate = sessionSampleRate
         self.onErrorSampleRate = onErrorSampleRate
         self.maskAllText = maskAllText
         self.maskAllImages = maskAllImages
         self.enableViewRendererV2 = enableViewRendererV2
         self.enableFastViewRendering = enableFastViewRendering
+
+        // These are the default values for the properties.
+        // Do not add any default values at the property declaration, because they will be overridden by the initializer.
```

Comment:
`m`: Now looking at this again, I'm reconsidering my statement yesterday. I think having the default values in one place as suggested in https://github.com/getsentry/sentry-cocoa/pull/5214, which has the downside of not showing the default values in the code docs, is easier to maintain. The current approach can break easily, but we're also not going to change this often. We can work around the code doc default value problem, by adding code docs for the initializers and the properties. The SentryOptions are written in ObjC and you can also only get the default value by looking at the code docs. I have no strong opinion on this one. I let you decide @philprime.

Reply:
We are trading maintainability for usability. I'll create a well-documented defaults type so that users can also easily understand the values. I will NOT add the default values to the code documentation for this initializer, because then we are back at the original issue of having multiple fields to maintain.

Reply:
I decided to add a comment to each initializer and property pointing to the default values type.